monai/data/dataset.py

```diff
@@ -883,8 +883,8 @@ class SmartCacheDataset(Randomizable, CacheDataset):
     def __init__(
         self,
         data: Sequence,
-        transform: Union[Sequence[Callable], Callable],
         replace_rate: float,
+        transform: Optional[Union[Sequence[Callable], Callable]] = None,
```

Comment:
this changes the argument list and potentially break external codes, perhaps we can give `replace_rate` a default value and remain the previous arg list? cc @Nic-Ma

Reply:
I added set the default value for `replace_rate` as 0.5 for now. @Nic-Ma let me know if any other value is more appropriate.
We can also remove the default value for transform but I think it would be nice to have a default value for `replace_rate` from new user's perspective so that they can just start using it without thinking about it.

Reply:
Hi @drbeh ,

For regular training cases, according to my experience, I never used a `replace_rate` greater than `0.2`. `0.5` may cause the training hard to converge?
Maybe default to `0.1` to make the training more stable, just like the random `prob` in all the transforms:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/intensity/array.py#L95
I am not sure, what's your `replace_rate` in the pathology task?
@dongyang0122 Do you have any best practice for the `replace_rate` in SmartCache?

Thanks in advance.

Reply:
@Nic-Ma @drbeh We used to use replace_rate 0.2 and it worked fine. In general, a small replace_rate will give you better model convergence since the model sees similar data at each training epoch. A larger replace_rate will lead to larger variance in the training process, and the convergence will be slightly worse.

Reply:
@dongyang0122 @Nic-Ma thanks for your input. I agree in the pathology tumor detection use case, we also used small values, so I set the default value to `0.1`.